### DISCUSSION TITLE
@xstate-inspect in sandbox

### DISCUSSION THREAD

**Discussion by JayNaire** [statelyai/xstate#1960] [Q&A] [answered]
Noob question: How can I stop my (vue template derived) sandboxes from refreshing automatically?

I only want to update (manually) when I've finished editing, say, index.ts; not after every keypress or so.
https://codesandbox.io/s/practical-shape-4dxze

Apart from ripping the focus away from the code when the visualisation refreshes, it produces multiple inspectors eg: Machine (x:7) instead of Machine (x:1) and duplicate reported events / sequences.

At the moment I put a deliberate error at the beginning of index.ts when I start editing and remove it when done. It gets tedious when trying out quick mods!

**Comment by ycaptain** [comment, marked as answer]
Press [cmd + ,] to open settings. Then switch `files.autoSave` to off and it works.
![image](https://user-images.githubusercontent.com/25996826/109418399-2a69cf00-7a03-11eb-9d5b-7d680734f6f7.png)

**Reply by JayNaire** [reply]
Many thanks for that. Sadly, it didn't work for me. Autosave was already off so saving the file isn't the trigger to refresh the visualisation - some "linty thing" is watching  keystrokes somewhere else I imagine.

**Reply by ycaptain** [reply]
For preview refresh, toggle two settings in File -> Preferences -> CodeSandbox Settings -> Preview.
Turn off "Preview on edit"
Turn off "Instant preview"
![image](https://user-images.githubusercontent.com/25996826/109449553-bde2e480-7a83-11eb-92ab-11886f89fe18.png)

**Reply by JayNaire** [reply]
That did it - thank you very much indeed!
